filecoin/minermonitor.go

```diff
@@ -0,0 +1,98 @@
+package filecoin
+
+import (
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type minerStatus struct {
```

Comment:
non-blocking, but this is the sort of package you can can write a unit test for pretty quickly. Your logic looks really good (I think), but it's still the sort of thing that might have weird edge cases you're not catch.

Reply:
clearly, I remain the only programmer at PL who likes unit tests.

Reply:
> clearly, I remain the only programmer in PLN who likes unit tests.

Be the change you want to see in the world Hannah.

Reply:
truthfully, and secretely, peerIDs are just strings -- 

here's code I use in another package for it:

```golang
var peerSeq int

// GeneratePeers creates n peer ids.
func GeneratePeers(n int) []peer.ID {
	peerIds := make([]peer.ID, 0, n)
	for i := 0; i < n; i++ {
		peerSeq++
		p := peer.ID(fmt.Sprint(peerSeq))
		peerIds = append(peerIds, p)
	}
	return peerIds
}
```

I wish packages that define these types would provide mocking generators. Sadly, this is not the case and I've encountered philosophical objections when I bring it up.